### PR TITLE
Always create Push-Challenge if polling is enabled

### LINKED ISSUE #2438: Create challege even if message could not be submitted to firebase
Can you check that a challenge is generated for a push token even if privacyidea was not able to submit the message to the firebase service?
This happens if the firebase token of the app has changed, but using push-poll, the app is still able to respond to the challenge without using firebase at all.

### PR DESCRIPTION
**Pull request by plettich**
Even if submitting the message for the Push-notification to the firebase
service fails a challenge is created if polling is enabled (either
globally or for that token specifically).
The App is then able to poll for open challenges and answer them
correctly.

Fixes #2438

### CODE REVIEW

File: privacyidea/lib/tokens/pushtoken.py [hunk 1]
```diff
@@ -729,6 +729,32 @@ def create_challenge(self, transactionid=None, options=None):
                                                      challenge, fb_gateway,
                                                      pem_privkey, options)
             res = fb_gateway.submit_message(self.get_tokeninfo("firebase_token"), smartphone_data)
+
+            # Create the challenge in the challenge table regardless of a failure
+            # when sending the Push message to the Firebase API.
+            # But only if polling is allowed.
+            allow_polling = get_action_values_from_options(
+                SCOPE.AUTH, PUSH_ACTION.ALLOW_POLLING, options=options) or PushAllowPolling.ALLOW
+            if (allow_polling == PushAllowPolling.ALLOW or
```
Comment (cornelinux) [inline privacyidea/lib/tokens/pushtoken.py:738]: Maybe I am mislead here. But we also need to create a challenge in the database if the message is successfully submitted to firebase. Even if not polling is allowed. 
(Which would be the default behaviour. We sill need the challenge, in the database for the token to work correctly like before.)

So to my understanding this should be

~~~~python
if (polling-conditions) or (res):
~~~~

**Review by cornelinux — CHANGES_REQUESTED** [review]
I think authentication will always fail, if polling is not allowed!

File: privacyidea/lib/tokens/pushtoken.py (hunk 1 above)
Comment (plettich) [reply privacyidea/lib/tokens/pushtoken.py:738]: You are absolutely right, good thing we have a review process :-)

Comment (plettich) [reply privacyidea/lib/tokens/pushtoken.py]: Done in cb4c8fb

File: privacyidea/lib/tokens/pushtoken.py [hunk 2]
```diff
@@ -729,6 +729,32 @@ def create_challenge(self, transactionid=None, options=None):
                                                      challenge, fb_gateway,
                                                      pem_privkey, options)
             res = fb_gateway.submit_message(self.get_tokeninfo("firebase_token"), smartphone_data)
+
+            # Create the challenge in the challenge table regardless of a failure
```
Comment (cornelinux) [inline privacyidea/lib/tokens/pushtoken.py:733]: The if-statement actually now means s.th. like

Create the challenge in the challenge table if either the messages was successfully submitted
to the Firebase API or if polling is allowed in general or for this specific token.

**Review by cornelinux — CHANGES_REQUESTED** [review]
Only nitpicking about the comment...

File: privacyidea/lib/tokens/pushtoken.py (hunk 2 above)
Comment (plettich) [reply privacyidea/lib/tokens/pushtoken.py]: Done in 012b6af7

**Review by cornelinux — APPROVED** [review]

**Pull request merged**